cmd/kubeadm/app/constants/constants.go

```diff
@@ -555,3 +552,36 @@ func GetDNSVersion(dnsType kubeadmapi.DNSAddOnType) string {
 func GetKubeletConfigMapName(k8sVersion *version.Version) string {
 	return fmt.Sprintf("%s%d.%d", KubeletBaseConfigurationConfigMapPrefix, k8sVersion.Major(), k8sVersion.Minor())
 }
+
+// GetCurrentKubernetesVersion returns current Kubernetes version supported by kubeadm
+func GetCurrentKubernetesVersion() *version.Version {
```

Comment:
to make this "constants" are you going to hide the values behind public `get` functions? incurring the same huge PR anyhow? I am no golang master so I might just be missing some golang feature here... Ether way, I like the idea of calculating them just once!

Reply:
It's in quotes, so these are going to be technically vars (but vars that never change).

Reply:
I personally like `calculate values` -> dump into package private vars -> expose via functions (accessor methods). But I will defer to @rosti 's greater experience with this project.

Reply:
@Klaven that's always better, however it will require, that this PR touch almost all of the files under `./cmd/kubeadm`. We can split it in a couple of commits (and PRs even), but this is low priority right now.

Reply:
cool. makes since! thanks!